### PR TITLE
[PreviewCard] Display PreviewCard view, add rounded box inside view

### PR DESCRIPTION
**Pull request by lewist13**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

This PR introduces the SwiftUI implementation of a Preview Card.
Included within the view is a rounded, bordered box. The box's properties are defined with non-tokenized values and tentative for now. 

Questions for the design/general:

1. Center the box?
2. ~~Move the box title below the box instead of inside?~~
3. Move the Constants variables into another file?

### Verification

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| all controls2 | previewcard_box_display_updated resized |

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/963)

### CODE REVIEW

File: ios/FluentUI.Demo/FluentUI.Demo.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -292,8 +294,8 @@
 				FD7254EC21471A3F002F4069 /* DateTimePickerDemoController.swift */,
 				A5DCA75D211E3A92005F4CB7 /* DrawerDemoController.swift */,
 				B4E8F65121CD9579008A1598 /* HUDDemoController.swift */,
-				5328D97926FBA3E900F3723B /* IndeterminateProgressBarDemoController.swift */,
```
Comment (harrieshin) [inline ios/FluentUI.Demo/FluentUI.Demo.xcodeproj/project.pbxproj:295]: that's weird. not sure why this change was made?

File: ios/FluentUI.Demo/FluentUI.Demo.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -275,10 +277,10 @@
 		A5CEC23220E452B80016922A /* Demos */ = {
 			isa = PBXGroup;
 			children = (
-				532FE3D926EA6D8D007539C0 /* ActivityIndicatorDemoController.swift */,
 				532FE3DA26EA6D8D007539C0 /* ActivityIndicatorDemoController_SwiftUI.swift */,
-				5373D55E2694C3070032A3B4 /* AvatarDemoController.swift */,
```
Comment (harrieshin) [inline ios/FluentUI.Demo/FluentUI.Demo.xcodeproj/project.pbxproj:280]: same here what's changing here?

File: ios/FluentUI.Demo/FluentUI.Demo/Demos/PreviewCardDemoController.swift [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        preconditionFailure("init(coder:) has not been implemented")
+    }
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        let previewCard = MSFPreviewCard(theme: nil)
+        let previewCardView = previewCard.view
+        view.addSubview(previewCardView)
+        previewCardView.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            previewCardView.topAnchor.constraint(equalTo: self.view.topAnchor, constant: 16.0),
```
Comment (harrieshin) [inline ios/FluentUI.Demo/FluentUI.Demo/Demos/PreviewCardDemoController.swift:24]: do we need "self" here? it doesn't seem ambiguous to me.

File: ios/FluentUI.Demo/FluentUI.Demo/Demos/PreviewCardDemoController.swift [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+            previewCardView.leadingAnchor.constraint(equalTo: self.view.leadingAnchor, constant: 16.0),
```
Comment (harrieshin) [inline ios/FluentUI.Demo/FluentUI.Demo/Demos/PreviewCardDemoController.swift:25]: might as well have 16 declare as static const

File: ios/FluentUI/PreviewCard/PreviewCard.swift [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        hostingController.disableSafeAreaInsets()
+    }
+    var window: UIWindow? {
+        return self.view.window
+    }
+    private var hostingController: FluentUIHostingController!
+    private var previewCardView: PreviewCard!
+}
+
+public struct PreviewCard: View {
```
Comment (harrieshin) [inline ios/FluentUI/PreviewCard/PreviewCard.swift:26]: lets start adding light documentation on this public class!

File: ios/FluentUI/PreviewCard/PreviewCard.swift [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+    @Environment(\.theme) var theme: FluentUIStyle
+    @Environment(\.windowProvider) var windowProvider: FluentUIWindowProvider?
+    private struct Constants {
+        static let boxMinWidth: CGFloat = 343
+        static let boxMinHeight: CGFloat = 200
+        static let boxRoundedTitle: String = "Rounded Box"
```
Comment (harrieshin) [inline ios/FluentUI/PreviewCard/PreviewCard.swift:33]: might as well remove this? we don't need a text right now. we don't want unlocalized text in our component code base anyway

File: ios/FluentUI/PreviewCard/PreviewCard.swift [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    private var hostingController: FluentUIHostingController!
+    private var previewCardView: PreviewCard!
+}
+
+public struct PreviewCard: View {
+
+    @Environment(\.theme) var theme: FluentUIStyle
+    @Environment(\.windowProvider) var windowProvider: FluentUIWindowProvider?
+    private struct Constants {
+        static let boxMinWidth: CGFloat = 343
```
Comment (harrieshin) [inline ios/FluentUI/PreviewCard/PreviewCard.swift:31]: are you adding todos to actually update with real sizes? i am kind of surprised this width is not even numbers.

File: ios/FluentUI.Demo/FluentUI.Demo/Demos/PreviewCardDemoController.swift (hunk 4 above)
Comment (mischreiber) [reply ios/FluentUI.Demo/FluentUI.Demo/Demos/PreviewCardDemoController.swift:25]: Yep, seeing a repeated number like this is a good sign that it can be stored in a variable somewhere, probably a static const like @harrieshin mentioned.

File: ios/FluentUI.Demo/FluentUI.Demo/Demos/PreviewCardDemoController.swift [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+
+class PreviewCardDemoController: DemoTableViewController {
+    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
+        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
+    }
+    required init?(coder: NSCoder) {
+        preconditionFailure("init(coder:) has not been implemented")
+    }
+// MARK: - Constraint Constant Attribute Values
+    private struct Constants {
```
Comment (harrieshin) [inline ios/FluentUI.Demo/FluentUI.Demo/Demos/PreviewCardDemoController.swift:18]: i believe we still try to declare private struct in the bottom of the class? https://microsoft.github.io/swift-guide/FileOrganization.html

File: ios/FluentUI/PreviewCard/PreviewCard.swift [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+
+@objc open class MSFPreviewCard: NSObject, FluentUIWindowProvider {
+    @objc open var view: UIView {
+        return hostingController.view
+    }
+    @objc public init(theme: FluentUIStyle?) {
+        super.init()
+        previewCardView = PreviewCard()
+        hostingController = FluentUIHostingController(rootView: AnyView(previewCardView.windowProvider(self)))
+        hostingController.disableSafeAreaInsets()
```
[the remark this answers is not in the document]
Comment (harrieshin) [reply ios/FluentUI/PreviewCard/PreviewCard.swift:22]: this branch is on top of vnext-token branch right? so we need a new merge ?

File: ios/FluentUI/PreviewCard/PreviewCard.swift [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+    }
+    /// Creates the PreviewCard.
+    public var body: some View {
+        innerContents
+            .background(
+                RoundedRectangle(cornerRadius: Constants.boxRadius)
+                    .strokeBorder(Color(.black), lineWidth: Constants.boxLineWidth)
+                    .background(
+                        RoundedRectangle(cornerRadius: Constants.boxRadius)
+                            .fill(.white)
```
Comment (harrieshin) [inline ios/FluentUI/PreviewCard/PreviewCard.swift:60]: this needs a todo for actually update dark and light mode

File: ios/FluentUI/PreviewCard/PreviewCard.swift [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+/// View that represents the PreviewCard.
+public struct PreviewCard: View {
+
+    @Environment(\.theme) var theme: FluentUIStyle
+    @Environment(\.windowProvider) var windowProvider: FluentUIWindowProvider?
+// MARK: - PreviewCard Box Properties
+    // TODO: - Update the hard values with tokens
+    private struct Constants {
+        static let boxMinWidth: CGFloat = 343
+        static let boxMinHeight: CGFloat = 147
```
Comment (harrieshin) [inline ios/FluentUI/PreviewCard/PreviewCard.swift:40]: lets not use "box" maybe cardMinHeight, cardRadius, cardBorderLIneWidth?

File: ios/FluentUI/PreviewCard/PreviewCard.swift [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+/// View that represents the PreviewCard.
+public struct PreviewCard: View {
+
+    @Environment(\.theme) var theme: FluentUIStyle
+    @Environment(\.windowProvider) var windowProvider: FluentUIWindowProvider?
+// MARK: - PreviewCard Box Properties
+    // TODO: - Update the hard values with tokens
+    private struct Constants {
```
Comment (harrieshin) [inline ios/FluentUI/PreviewCard/PreviewCard.swift:38]: also move the strcut to bottom?

File: ios/FluentUI.Demo/FluentUI.Demo/Demos/PreviewCardDemoController.swift (hunk 8 above)
Comment (lewist13) [reply ios/FluentUI.Demo/FluentUI.Demo/Demos/PreviewCardDemoController.swift:18]: Got it, thanks for the resource

File: ios/FluentUI/PreviewCard/PreviewCard.swift (hunk 11 above)
Comment (harrieshin) [reply ios/FluentUI/PreviewCard/PreviewCard.swift:40]: sorry cardCornerRadius?

**Review by harrieshin — APPROVED** [review]
